### ISSUE TITLE
Cannot update parameters of existing application in Marathon 1.4.0

### ISSUE THREAD

**Issue by naag** [mesosphere/marathon#5215]
Hi guys,

using Marathon 1.4.0 on Mesos 1.1.0, we're unable to update parameters of existing applications. We're using Docker on our Mesos slaves to run our applications. Each Docker image is tagged with it's git commit, so if we push out a new release and want to deploy it, we simply update the git commit inside the app definition at `container.docker.image` using a PUT request (we send the full application definition along).

Since Marathon 1.4.0, this now longer works. The previous PUT requests result in status code 422:

```
curl -w"\n" -f -XPUT -H"Content-Type: application/json" -d@production.json http://mesos-master1.domain:8080/v2/apps/some-api/production

curl: (22) The requested URL returned error: 422 Unprocessable Entity
```

 A PATCH request works though, but only if we send it directly to the Marathon leader. When sending it to one of the other nodes in a HA setup, no deployment is triggered.

This is our version detail:

```
{
  "name": "marathon",
  "version": "1.4.0",
  "buildref": "unknown",
  "elected": true,
  "leader": "mesos-master1.domain:8080",
  "frameworkId": "61322be0-8756-418c-a1b2-a27e5c8763e1-0000",
  "marathon_config": {
    "master": "zk://192.168.1.10:2181,192.168.1.11:2181,192.168.1.12:2181,192.168.1.13:2181,192.168.1.14:2181/mesos",
    "failover_timeout": 604800,
    "framework_name": "marathon",
    "ha": true,
    "checkpoint": true,
    "local_port_min": 10000,
    "local_port_max": 20000,
    "executor": "//cmd",
    "hostname": "mesos-master1.domain",
    "webui_url": null,
    "mesos_role": null,
    "task_launch_timeout": 300000,
    "task_reservation_timeout": 20000,
    "reconciliation_initial_delay": 15000,
    "reconciliation_interval": 60000,
    "mesos_user": "root",
    "leader_proxy_connection_timeout_ms": 5000,
    "leader_proxy_read_timeout_ms": 10000,
    "features": [],
    "mesos_leader_ui_url": "http://mesos-master1.domain:5050/"
  },
  "zookeeper_config": {
    "zk": "zk://192.168.1.10:2181,192.168.1.11:2181,192.168.1.12:2181,192.168.1.13:2181,192.168.1.14:2181/marathon",
    "zk_timeout": 10000,
    "zk_session_timeout": 10000,
    "zk_max_versions": 50
  },
  "event_subscriber": null,
  "http_config": {
    "http_port": 8080,
    "https_port": 8443
  }
}
```

OS Info:

```
Linux mesos-master1 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2+deb8u3 (2016-07-02) x86_64 GNU/Linux
```

Going from PUT to PATCH is not a problem for us, but the "need to send to leader" thing is a little odd. Until now we we're thinking that Marathon proxies everything to the master node itself.

**Comment by janisz** [comment]
Duplictes: https://github.com/mesosphere/marathon/issues/5211

**Comment by meichstedt** [comment]
Fixed on 1.4 via e1b79523741bf1c266b953eba1bcf8783f819c3a
